src/plots/plots.js

```diff
@@ -1979,8 +1981,16 @@ plots.doAutoMargin = function(gd) {
 
     var gs = fullLayout._size;
     var margin = fullLayout.margin;
+    var reservedMargins = {'t': 0, 'b': 0, 'l': 0, 'r': 0};
     var oldMargins = Lib.extendFlat({}, gs);
 
+    // We're assuming that each entry only pushes the margin out on one side
+    var margins = gd._fullLayout._reservedMargin;
+    for(var key in margins) {
+        var side = Object.keys(margins[key])[0];
+        var val = Object.values(margins[key])[0];
+        reservedMargins[side] += val;
+    }
```

Comment:
```suggestion
    var margins = gd._fullLayout._reservedMargin;
    for(var key in margins) {
        for(var side in margins[key]) {
            var val = margins[key][side];
            reservedMargins[side] = Math.max(reservedMargins[side], val);
        }
    }
```
- No reason to impose the limitation of only one side - I could easily see an object pushing out both top and right, for example. And if there's only one, adding this loop is no slower, might even be faster.
- Right now you only have one so this doesn't matter, but when there are multiple if we want them to push each other around (and sum their reserved pixels) we'll need some other mechanism. Until then they won't add, you'll just take the largest one. For example you might put a title and legend both container-positioned on top, with the title on the left and the legend on the right.

Reply:
Agreed, this makes sense!